Adds option to disable the PaperWM topbar workspace name indicator/button

jtaala opened the pull request:
This PR adds a user-settable option which disables the workspace indicator.

Fixes #387 #433.

From this PR's changes to `README.md`:

> If you prefer to use another workspace indicator (or prefer none at all), you can hide this workspace name element from Gnome topbar by executing the following command from a terminal:
> 
> ```
> dconf write /org/gnome/shell/extensions/paperwm/show-workspace-indicator false
> ```

_NOTE: this PR has been implemented in the [PaperWM-redux](https://github.com/PaperWM-redux/PaperWM) fork, which you can install if you want this, or any of [my PRs](https://github.com/paperwm/PaperWM/pulls/jtaala) that are open._

jtaala committed:
Added gschema option to show workspace indicator.

jtaala committed:
Connected show-workspace-indicator signnal and implemented methods to
show/hide both topbar element and space elements.

jtaala committed:
Refactor to more appropriately name the workspace indicator element
(from the ambiguous "labelWrapper".

jtaala committed:
Added function to update space focusModeIcon positions when workspace
indicator shows/hides.

jtaala committed:
Updated README.md.

Lythenas commented on topbar.js:
```diff
@@ -799,16 +806,34 @@ function fixTopBar() {
 }
 
+function fixWorkspaceIndicator() {
+    prefs.show_workspace_indicator ? menu.show() : menu.hide();
```
This should just be an `if`, right?

jtaala replied:
well, that's an inline if.  But yes, it's an if.

jtaala replied:
I might have misunderstood your question(?).  It needs to be an `if-else`, since if we change the `show-workspace-indicator` option to false, we want it to `hide()`.

Hence the inline if(else) statement.

Let me know if that didn't answer the question.

Lythenas commented on topbar.js:
```diff
@@ -799,16 +806,34 @@ function fixTopBar() {
+ * of the focusButton (mimics it).
+ * @param {Number} timeout needed to allow update before querying position.
+ */
+function setSpaceFocusModeIconPositions(timeout=0) {
```
Do we need a specific timeout here? Or does `0` always work?

jtaala replied:
default is 0, but a small timeout is needed when workspace indicator emits `show` and `hide` signals.

See https://github.com/paperwm/PaperWM/pull/513/files#diff-b0cab10735fabfe8a6e4455997859f505dbbe12a813ba3227c2952ba5a89ecddR671-R672

jtaala replied:
Yes, `0` always works, hence set it as default.

We also could do something like

```
function setSpaceFocusModeIconPositions(timeout) {
    timeout = timeout ?? 0;
    ...
}
```
Which is kind of equivalent (but maybe preferred since it would also cover the case where a developer might do a `setSpaceFocusModeIconPositions(undefined)` or `setSpaceFocusModeIconPositions(null)`) - I can change that one.

jtaala committed:
Small improvement to pattern used for `setSpaceFocusModeIconPositions`
method.

jtaala committed:
Merge remote-tracking branch 'upstream/develop' into option-disable-workspace-indicator

jtaala committed:
Added new setting decription for `show-workspace-indicator` (given
`README.md` changes from #514.

jtaala committed:
Fixed small typo.

jtaala commented:
Thanks @Lythenas for approving!

Note, I've just updated `README.md` and added this property to the new properties section from #514.  Please approve again when you have a chance and I'll merge this in. 

Thanks!

ChaiAK commented:
Hi, what happened to this, it's seemingly no longer a feature or in the readme?

jtaala commented:
@ChaiAK 

![image](https://github.com/user-attachments/assets/7b26c3f8-7aeb-4fe5-b2d8-cf2eb1af7039)

Enable that option to replace (and disable) workspace name indicator with standard gnome pill.

jtaala commented:
It's also in the readme:
https://github.com/paperwm/PaperWM?tab=readme-ov-file#the-workspace-stack--monitors

![image](https://github.com/user-attachments/assets/3ca24342-5391-4b80-bbc3-32ff3109dd00)
